File: ackermann.rb
def ackermann(m, n)
  if m === 0
    n + 1
  elsif m > 0 && n === 0
    ackermann m - 1, 1
  else m > 0 && n > 0
       ackermann m - 1, ackermann(m, n - 1)
  end
end

puts ackermann 4, 2

File: ackermann.rs
fn ackermann(m: usize, n: usize) -> usize {
    match (m, n) {
        (0, n) => n + 1,
        (m, 0) => ackermann(m - 1, 1),
        _ => ackermann(m, n - 1),
    }
}

fn main() {
    println!("{}", ackermann(4, 2));
}
